sources/api/prairiedog/src/bootconfig.rs

```diff
@@ -64,19 +64,28 @@ fn serialize_boot_settings_to_boot_config(boot_settings: &BootSettings) -> Resul
     // Preallocate string buffer to avoid a bunch of memory allocation calls when we append to
     // the string buffer
     let mut output = String::with_capacity(128);
-    if let Some(kernel_param) = &boot_settings.kernel_parameters {
-        for (key, values) in kernel_param.iter() {
-            output.push_str(&format!("kernel.{key}"));
+    // Output init parameters first since "init." comes before "kernel." alphabetically
+    if let Some(init_param) = &boot_settings.init_parameters {
+        // Sort keys alphabetically for consistent ordering
+        let mut sorted_keys: Vec<_> = init_param.keys().collect();
+        sorted_keys.sort_by_cached_key(|k| k.as_ref().to_string());
+        for key in sorted_keys {
+            let values = &init_param[key];
+            output.push_str(&format!("init.{key}"));
```

Comment:
Looks correct. You may be able to improve some of the density/reduce complexity of this function by deferring to a helper for writing the config values:

```rust
/// Writes parameters to output in sorted order with the given prefix
fn write_sorted_params(
    output: &mut String,
    params: &HashMap<BootConfigKey, Vec<BootConfigValue>>,
    prefix: &str,
) {
    let mut sorted_keys: Vec<_> = params.keys().collect();
    sorted_keys.sort_by_cached_key(|k| k.as_ref().to_string());

    for key in sorted_keys {
        let values = &params[key];
        output.push_str(&format!("{prefix}.{key}"));
        if !values.is_empty() {
            output.push_str(" =");
            append_boot_config_value_list(values, output);
        }
        output.push('\n');
    }
}
```